src/python/pants/backend/python/rules/black.py

```diff
@@ -36,19 +36,13 @@ class BlackSetup:
   resolved_requirements_pex: Pex
   merged_input_files: Digest
 
-  def generate_pex_arg_list(self, *, files: Set[str], check_only: bool) -> Tuple[str, ...]:
+  def generate_pex_arg_list(self, *, files: Tuple[str, ...], check_only: bool) -> Tuple[str, ...]:
     pex_args = []
     if check_only:
       pex_args.append("--check")
     if self.config_path is not None:
       pex_args.extend(["--config", self.config_path])
-    if files:
-      pex_args.extend(["--include", "|".join(re.escape(f) for f in files)])
-    # Black normally operates on all passed folders/files and traverses them recursively. We pass
-    # the directories we want and, crucially, use --include to ensure that Black only runs on the
-    # actual files we care about.
-    dirs = {f"{Path(filename).parent}" for filename in files}
-    pex_args.extend(sorted(dirs))
+    pex_args.extend(files)
```

Comment:
I'm very happy with this change if we're sure it doesn't break anything, but I think I may have had troubles when trying this initially, so would like to make sure this is well tested (maybe running a full codebase black on pants before and after and making sure they produce the same diff) before pushing it.

In particular, we're incidentally loosing the `re.escape` and I could see this causing issues.

Reply:
(Note: for the `re.escape`, it may make sense to add a test with some regex confusion going on to be sure to catch the problem if any)

Reply:
Yes, I'm confident this works. I tried running this against a file with unicode in the name and everything worked normally:

```
would reformat /private/var/.../src/python/pants/backend/python/rules/🔥.py
```

Additionally, I ran `./pants fmt-v2 src/python/pants::` against both master and this branch and they produced the same diff, minus the expected differences introduced in this PR.

I'm going to avoid adding a test because this is really Black functionality that they should be covering for us. If we pass `n` filenames to Black, we need to be able to assume that Black will run against those `n` files and nothing more or less. We also need to assume that Black will continue to support Unicode in file names - that is not something Pants should test.